### DISCUSSION TITLE
How to extract descriptors?

### DISCUSSION THREAD

**Discussion by RA-zoL** [deepmodeling/deepmd-kit#1393] [Q&A] [answered]
For a given molecule, is there any way to extract its concrete descriptors generated by a certain model?

**Comment by njzjz** [comment, marked as answer]
There is not an interface to do so. You need to modify some codes.
You can refer `deepmd.infer.DeepEval._eval_inner` and append `self._get_tensor("o_descriptor:0")` to `t_out`.
https://github.com/deepmodeling/deepmd-kit/blob/5a8479660a28f2fd31bafd2e5f3ab2a6549b7cfa/deepmd/infer/deep_pot.py#L325-L330

**Reply by RA-zoL** [reply]
Thank you very much!
